Gems/Atom/Feature/Common/Code/Source/PostProcessing/Fsr2TaaUpscalePass.cpp

```diff
@@ -0,0 +1,323 @@
+        // Negation is needed because we compute motion as (current clip position - previous clip position)
+        // but FSR2 expects the additive inverse.
+        m_fsr2DispatchDesc.motionVectorScale.x = aznumeric_cast<float>(m_fsr2DispatchDesc.renderSize.width);
+        m_fsr2DispatchDesc.motionVectorScale.y = aznumeric_cast<float>(m_fsr2DispatchDesc.renderSize.height);
+
+        // Determine the jitter offset based on current render scale. The sequence length grows as a function
+        // of the upscale amount. The FSR2 provided jitter sequence is a Halton sequence (outputs are in
+        // unit pixel space).
+        int jitterPhaseCount = ffxFsr2GetJitterPhaseCount(m_fsr2DispatchDesc.renderSize.width, m_fsr2ContextDesc.displaySize.width);
+        ffxFsr2GetJitterOffset(&m_fsr2DispatchDesc.jitterOffset.x, &m_fsr2DispatchDesc.jitterOffset.y, ++m_frameCount, jitterPhaseCount);
+
+        // Transfer from [-0.5, 0.5] to [-1/w, 1/w] in x and [-1/h, 1/h] in y
+        Vector2 clipSpaceJitter{ 2.f * m_fsr2DispatchDesc.jitterOffset.x / m_fsr2DispatchDesc.renderSize.width,
+                                 -2.f * m_fsr2DispatchDesc.jitterOffset.y / m_fsr2DispatchDesc.renderSize.height };
+
+        RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+        view->SetClipSpaceOffset(-clipSpaceJitter.GetX(), -clipSpaceJitter.GetY());
+        // FSR2 expects negative velocities in the motion vector buffer (current - previous clip position)
+        view->SetMotionVectorScale(-1.f, -1.f);
+
+        Vector2 nearFar = view->GetClipNearFar();
+        // TODO: Fix this when the near and far planes are actually fixed in the view
+        m_fsr2DispatchDesc.cameraNear = nearFar.GetY();
+        m_fsr2DispatchDesc.cameraFar = nearFar.GetX();
+
+        const AZ::Matrix4x4& viewToClip = view->GetViewToClipMatrix();
+        // TODO: this is an ugly way to compute FOV (assumes perspective RH view-to-clip matrix construction)
+        float viewToClip_2_2 = viewToClip.GetRow(1).GetY();
+        m_fsr2DispatchDesc.cameraFovAngleVertical = 2.f / AZ::Atan(viewToClip_2_2);
+
+        m_fsr2DispatchDesc.enableSharpening = r_fsr2SharpeningEnabled;
+        m_fsr2DispatchDesc.sharpness = r_fsr2SharpeningStrength;
+
+        double currentTimeMS = RHI::RHISystemInterface::Get()->GetCpuFrameTime();
+        m_fsr2DispatchDesc.frameTimeDelta = aznumeric_cast<float>(currentTimeMS - m_lastFrameTimeMS);
+        m_lastFrameTimeMS = currentTimeMS;
+
+        m_fsr2DispatchDesc.reset = false;
+
+        // TODO: Investigate if we preexposed values before writing to 16bit color target
+        m_fsr2DispatchDesc.preExposure = 1.f;
+    }
+
+    RPI::TimestampResult Fsr2TaaUpscalePass::GetTimestampResultInternal() const
+    {
+        return m_timestampResult;
+    }
+
+    void Fsr2TaaUpscalePass::SetupFrameGraphDependencies(RHI::FrameGraphInterface frameGraph)
```

Comment:
The void RenderPass::SetupFrameGraphDependencies(RHI::FrameGraphInterface frameGraph) should already do all these for you.